### ISSUE TITLE
[Bug]: Version 3.15.0 seems to be stuck in an infinite loop during "Checking folder changes". HDD is running hot.

### ISSUE THREAD

**Issue by chris-blues** [nextcloud/desktop#7613]
### ⚠️ Before submitting, please verify the following: ⚠️

- [x] This is a **bug**, not a question or a configuration issue.
- [x] This issue is **not** already reported on Github (I've searched it).
- [x] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

Debian 12 Linux, Gnome DE, Nextcloud Desktop client 3.15.0

Version 3.15.0 seems to be stuck in an infinite loop during "Checking folder changes". HDD is running hot. I've let it run for 30 minutes and more, just to see if it eventually stops at some point. It doesn't.

General startup of the client seems fine. File sync runs through fine. After that it goes into "Checking folder changes" - that's where it gets stuck in the loop. Reaction times get very slow on every action I take in the client. Gnome offers generic "App not responding, should I kill it or wait" warning.  With every older Client (e.g. 3.14.3) it works just fine. So there seems to be some regression introduced in 3.15.0...

If there's any more info I can provide, please let me know.

### Steps to reproduce

Startup Nextcloud-3.15.0-x86_64.AppImage
Wait for the app to start up...

### Expected behavior

normal startup without trashing my HDD...

### Which files are affected by this bug

folder

### Operating system

Linux

### Which version of the operating system you are running.

Debian 12

### Package

Official Linux AppImage

### Nextcloud Server version

30.0.3

### Nextcloud Desktop Client version

3.15.0

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 3.3.6 to 3.4.0)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [x] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Nextcloud Server logs

```shell

```

**Comment by chris-blues** [comment]
[Nextcloud_debug_archive.zip](https://github.com/user-attachments/files/18052266/Nextcloud_debug_archive.zip)

**Comment by Thorsten42** [comment]
I have the same issue on arch linux also with gnome. My nextcloud folder is located on a ntfs drive (windows dual boot) and mounted using ntfs-3g. I've reverted to 3.14.3 and everything worked again

**Comment by chris-blues** [comment]
Im also running on a NTFS drive! Funny, didn't think of that! Thanks for the input!

**Comment by MadFatVlad** [comment]
Same issue with NTFS partition on Arch Linux when using 3.15.0
3.14.3 works fine

**Comment by stefan-franz** [comment]
Same Problem on Linux Mint 22 Cinnamon with 3.15 Flatpak 

Hope this will fixed fast.

**Comment by chris-blues** [comment]
Desktop Client version 3.15.1 still has same issue.

[Nextcloud_debug_archive.zip](https://github.com/user-attachments/files/18128394/Nextcloud_debug_archive.zip)

**Comment by chris-blues** [comment]
Desktop Client version 3.15.2 still has same issue.

[Nextcloud_debug_archive.zip](https://github.com/user-attachments/files/18168826/Nextcloud_debug_archive.zip)

**Comment by blwh** [comment]
Can confirm I experience the same issues running 3.15.2 on Arch using KDE, ext4 formatted drives.

I tried downgrading to 3.14.3 but the problem still persists. If I restart nextcloud it will sync fine until I change a file - then it hangs.

**Comment by alraban** [comment]
I am also experiencing the same hanging issue on Arch Linux with only ext4 formatted drives.  Downgrading to 3.14.1 seems to resolve the issue, but every version after that one leads to UI hangs and seemingly indefinite CPU usage.  If others here could confirm whether 3.14.1 does or doesn't show the same issue, that might help bisect the issue.

**Comment by stefan-franz** [comment]
3.13.4 Appimage is running on my Linux Mint 22 the last, that runs normal. 
3.14 is not startable as an Appimage - wants passwords busshit
3.15.2 als flatpak (normal i use flatpak) has also the bug that it syncs all time, despite nothing is changed on the file system. 
3.14.x as flatpak had run fine. 3.15.x brought the bug

**Comment by petrkr** [comment]
Just updated on arch... Same result. CPU 100%, flooding logs.

I already deleted package cache. How I can download previous version at arch linux? On mirror is already new version.

**Comment by alraban** [comment]
Check the arch linux archive for prior versions: https://archive.archlinux.org/

**Comment by petrkr** [comment]
> Check the arch linux archive for prior versions: https://archive.archlinux.org/

Thanks https://archive.archlinux.org/repos/2024/11/09/extra/os/x86_64/nextcloud-client-2%3A3.14.3-1-x86_64.pkg.tar.zst

I tried to compile it from sources, but after hour it seg faulted in some test

**Comment by blwh** [comment]
> I am also experiencing the same hanging issue on Arch Linux with only ext4 formatted drives. Downgrading to 3.14.1 seems to resolve the issue, but every version after that one leads to UI hangs and seemingly indefinite CPU usage for me. It sounds like the issue only cropped up after 3.14.3 for some users, but I definitely see the same issue with that build, and it sounds like some others here also still saw issues with 3.14.3.
> 
> If others here could confirm whether 3.14.1 does or doesn't show the same issue, that might help bisect the issue.

This did not resolve the issue for me.

I did some testing. I find that if I create an empty file (`touch foo` for example) in the Nextcloud root folder or in a folder with fewer files there is no issue. However, if I create a file in a specific folder (a 130 GB one with many files and git repos) Nextcloud hangs as described by this issue.

**Comment by mbiebl** [comment]
I can confirm the problem (on Debian sid using 3.15.0).
My nextcloud folder is also on an NTFS partition and after the upgrade I saw a high CPU load of the nextcloud binary and the ntfs-3g fuse driver.
I ran git bisect which showed 5b2af166d3d9c8537c565922750392d4a3f6610e as the first faulty commit. 
I suppose that trying to apply the file permissions does not work on NTFS so nextcloud tries it over and over again.

**Comment by mbiebl** [comment]
@mgallien ^ can you please have a look

**Comment by mbiebl** [comment]
I've rebuilt 3.15.2 with 5b2af166d3d9c8537c565922750392d4a3f6610e reverted and nextcloud-desktop is working properly again.

**Comment by chris-blues** [comment]
Very nice! Now let's hope that someone from Nextcloud ever sees this and puts this to work. Thanks a lot.

I'd love to put a thumbs up or sth, but github won't let me in anymore since I stupidly activated 2 factor auth. Microsoft is very actively involved in locking people out...

Anyway! Good work! And thanks!

**Comment by mbiebl** [comment]
For completeness sake, the NTFS partition is mounted like this:
```
$ grep data /etc/fstab
LABEL=Data      /mnt/data       ntfs-3g auto            0       0

$ findmnt /mnt/data 
TARGET    SOURCE          FSTYPE  OPTIONS
/mnt/data /dev/nvme0n1p11 fuseblk rw,relatime,user_id=0,group_id=0,allow_other,blksize=4096
```

**Comment by mbiebl** [comment]
I've filed a downstream bug report at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1091614 and the Debian maintainer was so kind to revert the problematic commit in the Debian package until a proper fix is found.

**Comment by thieneret** [comment]
3.15.50 version from daily channel still has this bug. (OS: Linux Mint 22)

**Comment by petrkr** [comment]
> I've filed a downstream bug report at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1091614 and the Debian maintainer was so kind to revert the problematic commit in the Debian package until a proper fix is found.

Is possible do that same for other distros too?

Everytime when I update arch I have to downgrade afterwards nextcloud client, mostly I remember right after 100% cpu and battery drain after 1 hour instead 6h.

**Comment by chris-blues** [comment]
For the record: Version 3.15.3 still has the same issue...

I'm glad to see some movement! Thanks!

**Comment by mgallien** [comment]
@chris-blues if you can, could you test the AppImage once ready within https://github.com/nextcloud/desktop/pull/7745 ?
should have the fix

**Comment by chris-blues** [comment]
No problem. Does it appear later on  that page?

**Comment by chris-blues** [comment]
Finally it crashed...  

[Nextcloud_debug_archive.zip](https://github.com/user-attachments/files/18378247/Nextcloud_debug_archive.zip)

**Comment by mbiebl** [comment]
I can confirm. The AppImage still shows the same buggy behaviour

**Comment by mbiebl** [comment]
> I can confirm. The AppImage still shows the same buggy behaviour

My NTFS file system type is reported as `fuseblk` , not `NTFS`

**Comment by petrkr** [comment]
> > I can confirm. The AppImage still shows the same buggy behaviour
> 
> My NTFS file system type is reported as `fuseblk` , not `NTFS`

Also fuseblk can be anything baded on FUSE.

Like sshfs, ftpfs, webdavfs, etc...

**Comment by mbiebl** [comment]
It's probably better to check for required features then explicit file system types.

**Comment by petrkr** [comment]
But someone reported bug also on ext4 in comnents.

So I will not go only for ntfs-3g.

Later in week I can test it on second laptop, where  I did not updated yet and nextcloud storage is on btrfs with same content as on other laptop, where it is on ntfs-3g, but snce it was first updated system, i did not updated others.

**Comment by mbiebl** [comment]
> But someone reported bug also on ext4 in comnents.

Maybe different mount options or maybe it's a different issue which only looks like this one.

**Comment by chris-blues** [comment]
In _my_ /etc/fstab it says `ntfs`, if that should be relevant.

**Comment by mbiebl** [comment]
> So I will not go only for ntfs-3g.

I fear you might be right here. He have a host of fuse based file systems, file systems with non-UNIX semantics (FAT, NTFS etc). So having a blacklist / whitelist as in https://github.com/nextcloud/desktop/pull/7745#pullrequestreview-2542632747 is probably not gonna be maintainable.

**Comment by petrkr** [comment]
> In _my_ /etc/fstab it says `ntfs`, if that should be relevant.

I have ntfs-3g in fstab

When I'll get home I'll test update on that laptop with btrfs

**Comment by alraban** [comment]
> > But someone reported bug also on ext4 in comnents.
> 
> Maybe different mount options or maybe it's a different issue which only looks like this one.

There are several other issues reporting the desktop client freezing in a loop with high CPU usage even on Linux desktop for several versions.  I have been seeing the issue since 3.14.1 and only have ext4 file systems.  Downgrading to 3.14.1 or below fixes it entirely for me.  I think this issue is more than just a file-system specific issue (or if note there are two issues that present almost identically).

See: 
https://github.com/nextcloud/desktop/issues/7729
https://github.com/nextcloud/desktop/issues/7456
https://github.com/nextcloud/desktop/issues/7697

**Comment by chris-blues** [comment]
For the record:
On Debian systems the mount option 'ntfs' also loads the ntfs-3g driver since Squeeze.

**Comment by alraban** [comment]
To be clear, the first two of those bug reports was filed against the current version, not a previous version so the issue very much still exists.  I'm unconvinced that the issues are not being caused by the same root issue.  I don't have the technical skills to bisect the client, I'm just trying to provide the devs with relevant information.

**Comment by blwh** [comment]
> > > But someone reported bug also on ext4 in comnents.
> > 
> > 
> > Maybe different mount options or maybe it's a different issue which only looks like this one.
> 
> There are several other issues reporting the desktop client freezing in a loop with high CPU usage on Linux desktop for several versions. I have been seeing the issue on every client version after 3.14.1 and only have ext4 file systems. Downgrading to 3.14.1 or below fixes it entirely for me. I think this issue is more than just a file-system specific issue (or if not, there are two issues that present almost identically).
> 
> See: [#7729](https://github.com/nextcloud/desktop/issues/7729) [#7697](https://github.com/nextcloud/desktop/issues/7697) [#7456](https://github.com/nextcloud/desktop/issues/7456)

This is exactly my case as well. I am running 3.14.0 and it works just fine on my machines.

**Comment by chris-blues** [comment]
I too believe, we're dealing with multiple issues. This issue about NTFS drives clearly started with version 3.15.0. One of the other issues started earlier.

Then again, I'm not familiar with the code in question, so dismiss this as you see fit!

**Comment by mbiebl** [comment]
> I too believe, we're dealing with multiple issues. This issue about NTFS drives clearly started with version 3.15.0. One of the other issues started earlier.

The problematic commit I could identify is https://github.com/nextcloud/desktop/commit/5b2af166d3d9c8537c565922750392d4a3f6610e which you can see is part of 
v3.15.0 which matches what you are seeing. This particular commit has not been backported to the stable-3.14 branch. So yes, I think those other bug reports that were mentioned above have a different root cause.

**Comment by pati-ni** [comment]
Still the same on ext4 with the latest AppImage on arch linux. After some syncing, CPU gets to 100% and the client blocks (also the UI). Is the UI blocked for everyone else?

**Comment by chris-blues** [comment]
Yeah, very laggy after a short while.

**Comment by mgallien** [comment]
> It's probably better to check for required features then explicit file system types.

That was my first idea and I now get why I did the wrong choice (was not expecting fuse to be in use for example).

I will for now white list a couple of file system types that are matching the most common use cases for which the change was introduced. I will try to update the patch with some feature tests later (not sure when I will be able to introduce that).

**Comment by mgallien** [comment]
people please keep this on topic.
This issue is about some linux file system that have a weird semantic with UNIX style file permissions and the client trying in loop to fix them.
Other unrelated topics should go into a different issue.

**Comment by mbiebl** [comment]
> I will for now white list a couple of file system types that are matching the most common use cases for which the change was introduced. I will try to update the patch with some feature tests later (not sure when I will be able to introduce that).

That seems like a very sensible approach: get a fix out soon with a  whitelist for known working file systems (I expect this includes ext* and maybe btrfs and xfs) and have a more elaborate solution for later. Thanks @mgallien !

**Comment by petrkr** [comment]
> > I will for now white list a couple of file system types that are matching the most common use cases for which the change was introduced. I will try to update the patch with some feature tests later (not sure when I will be able to introduce that).
> 
> That seems like a very sensible approach: get a fix out soon with a whitelist for known working file systems (I expect this includes ext* and maybe btrfs and xfs) and have a more elaborate solution for later. Thanks [@mgallien](https://github.com/mgallien) !

As I tested, btrfs seems be fine.

**Comment by chris-blues** [comment]
It seems a solution is well on the way! Nice work everyone! And thanks!

**Comment by pati-ni** [comment]
> people please keep this on topic.
> This issue is about some linux file system types that have a weird semantic with UNIX style file permissions and the client trying in loop to fix them.

How do we know which issue affects us? It seems we have similar symptoms across different filesystems. Is an NTFS partition enough to contaminate the account and affect other filesystems that use the same account?

**Comment by petrkr** [comment]
> > people please keep this on topic.
> > This issue is about some linux file system types that have a weird semantic with UNIX style file permissions and the client trying in loop to fix them.
> 
> How do we know which issue affects us? It seems we have similar symptoms across different filesystems. Is an NTFS partition enough to contaminate the account and affect other filesystems that use the same account?

Probably not.

I have same account on two laptops. Both are Arch linux.
One use btrfs and 3.15.3 is fine
Other ude ntfs-3g and that does not work. But 3.14.3 is fine

**Comment by chris-blues** [comment]
> Is an NTFS partition enough to contaminate the account and affect other filesystems that use the same account?

This particular bug didn't compromise anything. It just didn't stop running the harddisk pretty hard, which is bad for the hardware in the long run. So, no data contamination etc whatsoever. About any other bug, I just don't know.

**Comment by grossardt** [comment]
I had the same(?) issue (non-responsive nextcloud-client using 100% cpu of one core, syncs fine after killing and restarting but goes back into hanging after the sync is complete) with 3.15.3 on arch. Nextcloud folder is on an ext4 VFS partition inside a dm-crypt, if that helps. No NTFS. Downgrade to 3.14.3 seems to have fixed the issue.
(May also be a case of #7729 if this is actually a different bug.)

**Comment by ArcticFall** [comment]
Because I have not read about this here yet: this also seems to happen on macOS with APFS.

**Comment by janschafrich** [comment]
I have the same issue. I'm using client 3.15.3 from the official nextcloud ppa on PopOs 22.04 with Gnome and ext4 formatted drives. I dual boot Windows on a separate NTFS partition every now and then where I also have a Nextcloud client installed. 
Otherwises I use the nextcloud client on iOS devices. 

```
user@host:~$ nextcloud --version
Warning: Ignoring WAYLAND_DISPLAY on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
nextcloud.gui.application: Migrating old config from "/home/jscha/.local/share/Nextcloud" to "/home/jscha/.config/Nextcloud"
nextcloud.gui.application: Failed to move the old config directory to its new location ( "/home/jscha/.local/share/Nextcloud" to "/home/jscha/.config/Nextcloud" )
nextcloud.gui.application: Will move the individual files QList("nextcloud_sync.log")
nextcloud.gui.application: Fallback move of  "nextcloud_sync.log" also failed
Nextcloud version 3.15.3-20250107.145041.a595d5bcb-1.0~jammy1
Using Qt 6.2.4, built against Qt 6.2.4
Using Qt platform plugin 'xcb'
Using 'OpenSSL 3.0.2 15 Mar 2022'
Running on Pop!_OS 22.04 LTS, x86_64
```

I installed 3.13.3 AppImage, here idle utilization is close to zero, as expected. Gonna stick to this one for now.

**Comment by geppi** [comment]
As others I'm experiencing the same issue with the Nextcloud client on Ubuntu 24.04 and the Nextcloud folder on an NTFS filesystem.

Could the mount options make a difference?
On my system they are: `/sbin/mount.ntfs /dev/mapper/data_veracrypt /data -o rw,uid=1000,gid=1000,nodev`

**Comment by chris-blues** [comment]
> Could the mount options make a difference?

I don't believe so! I think, the underlying problem is, that Linux/Unix/mac style systems have very different approach to file permissions compared to Windows system. Thus the permissions stuff doesn't reliably work. At least in this case, it fucks up the Nextcloud Client to set the file permissions on NTFS drive, resulting in an infinite loop, trying to correct the permissions.

Anyone with deeper knowledge, please correct me if I'm wrong. :o)

**Comment by pati-ni** [comment]
> 3.14.3 should work fine as well.

As @grossardt pointed out, 3.14.1 is the correct one to downgrade, at least for my case.  3.14.3 is not getting stuck like 3.15.x but eventually  after a 
 longer period it also hits 100% CPU.

**Comment by chris-blues** [comment]
Yeah, I'm guessing that we're dealing with 2 different issues here. I'm not sure though. But the fact, that there is 2 versions making a difference, points towards it. Yet, guessing doesn't help here anyway. Just my 2 pennies...

**Comment by luk-brue** [comment]
I do not have high CPU usage, but I do have the filesystem problem. If I sync to the ext4 partition, where the OS is stored as well, no problems. If I sync to an NTFS partition on a different drive, (that can also be accessed by Windows Dual Boot) - it keeps saying: "Looking for folder changes" or "preparing to sync" or "beginning sync" - or it is offline altogether even though I have internet. Although, sometimes it works briefly and even syncs some files that I added. But then it stops recognizing new files, even after hours of waiting. 

OS: Linux Mint 21.3 Cinnamon
Client: I built the #7745 Pull Request from Source, hoping it would save me (see my [Blog Post](https://lukbrue.quarto.pub/lb-blog/posts/nextcloud-desktop-linux-mint/)). I tried a lot of different clients, older ones, very old ones, but they were all stuck in Offline Mode. Maybe because of different issues with my setup. 

The fact that it does not work since a few months now painfully highlights how important Nextcloud sync has become in my daily life. Maybe you can take it as a compliment. Keep up the good work.

**Comment by affe42** [comment]
Can anyone confirm if v3.16 fixed anything regarding disk trashing & endless "sync"? Based on my tests, it didn't.

Oh well... back to v13.4.3.

**Comment by luk-brue** [comment]
@affe42 no, it does not fix the bug, that is why this bug is still open. They put it on the list for 3.16.1 if I remember correctly. You can track the progress here https://github.com/orgs/nextcloud/projects/117 
It is already "In Progress", which is a good sign.

**Comment by chris-blues** [comment]
Indeed, the problem persists! Too bad! Anyway here's my log, if it helps...

[Nextcloud_debug_archive.zip](https://github.com/user-attachments/files/19192460/Nextcloud_debug_archive.zip)

Back to 3.14.3...

**Comment by chris-blues** [comment]
3.16.1 still has same issue... Getting tired of cleaning up logs, so github's upload limit doesn't explode...

Additionally, since 3.16.0 the client messes up the sync_*.db's so everytime I try the new version, the old 3.14.3 version has to resync everything from scratch! That's quite a mess! Please don't fuck up my DBs people!

**Comment by petrkr** [comment]
I already asked arch linux if they will  downgrade package, but they said problem is not on their side, so they will not do anything and there are not so many user with ntfs

https://gitlab.archlinux.org/archlinux/packaging/packages/nextcloud-client/-/issues/10

**Comment by tomdereub** [comment]
It seems that we have the same bug on windows 10/11 (so ntfs), desktop client 3.15.3 and nextcloud server 30.0.6.
Client gets randomly stuck on "Checking folder changes", we are forced to quit and restart it, and sometimes it manages to finish sync.
![Image](https://github.com/user-attachments/assets/82a9d7c1-8312-4dde-96aa-56ddfbd7b084)

**Comment by grossardt** [comment]
I just wanted to reiterate that this bug is not exclusively an issue with NTFS partitions. I have no NTFS partition involved but experience the same problem. My nextcloud files are in an ext4 on LVM on LUKS partition.

**Comment by mbiebl** [comment]
> I just wanted to reiterate that this bug is not exclusively an issue with NTFS partitions. I have no NTFS partition involved but experience the same problem. My nextcloud files are in an ext4 on LVM on LUKS partition.

same symptoms, but different root cause. Thus should be handled via a separate bug report.

@mgallien can you please lock this issue? 
The root cause is pretty much known at this point and those me too's do not add any value and are just cluttering the bug tracker.

**Comment by chris-blues** [comment]
> Yes, lock it, do not solve it, make it as 'not planed' we will forever use 3.14.3 version.

That's not fair! There is a fix in progress, which just haven't been finished yet. I too am getting restless, since it's been quite a while, but no reason to be *that* sarcastic! Seems like the dev's need more personnel!

**Comment by chris-blues** [comment]
For the record: 3.16.2 still has same issue...

**Comment by jbfriedrich** [comment]
I can second that. Every time when I put the MacBook to sleep I still have this issue. I need to completely quit the client and restart for it to resolve.

**Comment by MystikReasons** [comment]
Can confirm that the issue still persists on 3.16.2 on Arch Linux stuck on high cpu consumption and being unresponsive. This is still a huge problem since it drains my laptop battery in no time. Normally I would get 10-11 hours with 100%-10%, with Nextcloud active in the unresponsive state, it reduces the battery to around 4 hours.

**Comment by Kleisli** [comment]
As a workaround I have paused sync for all instances and only activate it on demand.

**Comment by thejulian** [comment]
On MacOS Apple Silicon the client is also stuck at "preparing to sync" or "Checking Folder Changes"

**Comment by oppiman** [comment]
chiming in:

My NTFS drive is mounted:
/dev/disk/by-uuid/B208763A0875FDA3 /media/oppi/DATA auto nosuid,nodev,nofail,x-gvfs-show,uid=1000,gid=1000\040 0 0

All is fine up to client 3.14.3.
Anything later loops endless.
(using App Image, but tried flatpak as well - no difference)

My OS is latest LMDE.

**Comment by chris-blues** [comment]
2.16.3 still has the same issue...

**Comment by GetGeit** [comment]
Same issues here with 3.16.3 but server storage is ZFS in my case.

Running the 3.14.1 appimage without issue.

**Comment by MystikReasons** [comment]
Can confirm that the issue is still present with 3.16.3

![Image](https://github.com/user-attachments/assets/2ad50891-e281-4ebf-91c1-15ff8260cc4f)

**Comment by woah-buddy** [comment]
I, too, am experiencing this. No NTFS is involved in my setup either. Happy to troubleshoot/help however I can. This is not an issue with VFS disabled/the non-VFS client in use.

**Comment by chris-blues** [comment]
3.16.4 still has same issue...

@mgallien and @mbiebl is there any sense in keep on reporting this? Or is it wasting my time? Any info, on when the fix is coming is appreciated!
Thanks!

**Comment by lepolau** [comment]
same issue with nextcloud-client 3.16.4-1 on arch linux with kde plasma.
Revert to 3.14.3 (26658) solve the problem.

**Comment by oppiman** [comment]
FYI
it is solved for me.
See: https://help.nextcloud.com/t/endless-syncing-on-linux-3-15-2-and-3-15-3-3-14-3-was-fine/214719/37?u=micha_o

So no bug I guess, but a fundamental change in behavior that seemingly nobody is able to explain.

**Comment by chris-blues** [comment]
> ... is it clearer to the developers what is causing this behavior?

Yes, it's been clear for a while now. As I understand it, the client asks for the FS type, expecting 'NTFS' to be tha answer, but Linux reports it as 'fuseblk', which could be anything that is supported by fuse.

Though I'm surprised, that this still wasn't fixed...

**Comment by petrkr** [comment]
> > Yes, it's been clear for a while now. As I understand it, the client asks for the FS type, expecting 'NTFS' to be the answer, but Linux reports it as 'fuseblk', which could be anything that is supported by fuse.
> 
> Oh, thanks for this update! So, it is not purely a Nextcloud issue? Should we cross-post this to the fuse repo to see if they can fix it on their end?

It is not fuseblk issue.

Everyone use ntfs-3g which uses fuse. So it is not 'badly' reported as fuseblk. It is fuseblk.

Issue is definitely on nextcloud side, as this driver is used for years, because kernel one does not support write.

And also nextcloud client 3.14 works without issues with it.

**Comment by jbfriedrich** [comment]
I have this issue on macOS, without NTFS being involved at all 😔.

**Comment by muety** [comment]
Is there a potential fix for this in the offing? Being stuck with 3.14.3 also prevents me from upgrading my system to Fedora 42 meanwhile.

**Comment by oblivian** [comment]
I have the same problem on MacOS. Endless loop "Checking folder changes"
3.16.5 (and 3.16.4 after I downgraded. No luck).

**Comment by Thorsten42** [comment]
Hm, what does this removed from "desktop clients team" mean? Will someone still fix it or is it time to accept this issue?

**Comment by Thorsten42** [comment]
Has someone tried out using the newer ntfs3 kernel driver instead of the fuse ntfs-3g driver? It could circumvent this issue. 

edit: I actually switched to the ntfs3 kernel driver and it works with the newest nextcloud client.

**Comment by muety** [comment]
> [...] I actually switched to the ntfs3 kernel driver and it works with the newest nextcloud client.

Great to know! Did you experience any problems, as the driver was known to be somewhat buggy in the past? Also, can you briefly go into what was required to switch to the kernel driver? Simply changing `ntfs-3g` to `ntfs` in `/etc/fstab` yields permission issues (whole filesystem read-only), but admittedly I'm also not too much of a Linux pro.

**Comment by JohnDepon** [comment]
Same issue on Ubuntu 24.04 / KDE. No NTFS partitions at all.

**Comment by stefan-franz** [comment]
3.16.5 - not working on Linux Mint 22.1
Synchs endless, very slow, and takes huge resources. 

3.13.4 is the last version that works for me on Linux Mint 22.1

It seems no one is interested so repair that bug. Since 6 month the Nextcloud App with Linux makes problems. Was running many years fine and now not usable.   

If any dev wants to repair it and needs a log file or anything, i would provide. 
With Mixxx or Zoom i had problems with Mint the last 4 weeks. I got after reporting a special Debug version, installed, provided the log files and a few days later the bug was fixed. Not able here with Nextcloud?

**Comment by Thorsten42** [comment]
@muety: for read/write it needs to be ntfs3. According to the arch wiki just "ntfs" mimics the old-old-read-only ntfs driver and only allows read-only mount

I replaced my old fstab entry: 
```
UUID=idnr /media/thorsten/daten ntfs-3g uid=thorsten,gid=users 0 0
```

with (ntfs-3g -> ntfs3 and i added the nofail option)
```
UUID=idnr /media/thorsten/daten ntfs3 uid=thorsten,gid=users,nofail 0 0
```

I don't know how stable the driver is but at least the kernel docs state: "NTFS3 is fully functional NTFS Read-Write driver." (https://docs.kernel.org/filesystems/ntfs3.html)

The only issue i heard of but didn't occurred for me is that the driver refuses to mounts a volume if the "dirt" flag is not cleared, which happens if the drive is not unmounted due to a system crash for example. Therefore, i added the nofail flag. It's described in 3.3 in the arch wiki  https://wiki.archlinux.org/title/NTFS. The error message might sound a bit alarming but it should be fixable with ntfsfix -d /dev/drive_name or with chkdsk from windows

**Comment by chris-blues** [comment]
> [@muety](https://github.com/muety): 
> I replaced my old fstab entry:
> 
> ```
> UUID=idnr /media/thorsten/daten ntfs-3g uid=thorsten,gid=users 0 0
> ```
> 
> with (ntfs-3g -> ntfs3 and i added the nofail option)
> 
> ```
> UUID=idnr /media/thorsten/daten ntfs3 uid=thorsten,gid=users,nofail 0 0
> ```

Thanks a lot! I finally came around to try this out! And it works fine with NC-client 4.0.1 AppImage.

Though I really hope a fix will come around soon! 💪

**Comment by mgallien** [comment]
the PR to fix this has just been merged
if anyone is still willing to contribute a manual test (you can use the AppImage packages in the PR) , feel free to do it

**Comment by muety** [comment]
Thanks a lot for fixing this! Will give it a try later 👍

**Comment by mgallien** [comment]
sorry for the delay
the fix was more complex than anticipated

**Comment by oppiman** [comment]
> sorry for the delay the fix was more complex than anticipated

No need to be sorry, Thanks a lot for working on that.
Unfortunately, after an OS upgrade on all my systems, and a lot of fumbling with ntfs driver and fstab options, I cannot reproduce the issues I had back then - and I don't want to break stuff again...

What I'd like to understand: 
You wrote "fix was more complex".
But the issue got introduced with 3.15.
Shouldn't it be possible to identify what has changed from 3.14 3.15 and simply revert?

**Comment by chris-blues** [comment]
I just swithed to NTFS-3G driver, and all seems normal, running 4.0.4 AppImage.

It stays responsive, almost no CPU load, HDD silent. I would consider this bug fixed! Great work!

Thanks a lot for fixing this!
